src/main/java/seedu/duke/parsermanager/ParseCheckClient.java

```diff
@@ -0,0 +1,147 @@
+            int integer;
+            try {
+                integer = Integer.parseInt(detail);
+            } catch (NumberFormatException e) {
+                throw new NotIntegerException(EXCEPTION);
+            }
+            integerDetails.add(integer - 1); // Convert to 0-index
+        }
+        return integerDetails;
+    }
+
+    private void validateCheckClientDetails(ArrayList<Integer> checkClientDetails) throws InvalidIndexException {
+        int clientIndex = checkClientDetails.get(0);
+        checkForClientListIndexOutOfBounds(clientIndex);
+    }
+
+    private int[] getFlagIndexPositions(String commandDetail, String[] flags) {
+        int[] flagIndexPositions = new int[flags.length];
+
+        for (int i = 0; i < flags.length; i++) {
+            flagIndexPositions[i] = commandDetail.indexOf(flags[i]);
+        }
+        return flagIndexPositions;
+    }
+
+    private void checkForMissingFlags(int[] flagIndexPositions) throws MissingFlagException {
+        for (int flagIndex : flagIndexPositions) {
+            if (!isFlagPresent(flagIndex)) {
+                throw  new MissingFlagException(EXCEPTION);
+            }
+        }
+    }
+
+    private void checkFlagsOrder(int[] flagIndexPositions) throws IncorrectFlagOrderException {
+        for (int i = 0; i < flagIndexPositions.length - 1; i++) {
+            checkForCorrectFlagOrder(flagIndexPositions[i], flagIndexPositions[i + 1]);
+        }
+    }
+
+    private ArrayList<String> extractCommandDetails(String rawCommandDetail, String[] flags,
+                                                    int[] flagIndexPositions) {
+        ArrayList<String> extractedCommandDetails = new ArrayList<>();
+        for (int i = 0; i < flags.length; i++) {
+            String extractedDetail;
+            if (i == flags.length - 1) {
+                /* The extracted detail for the last flag starts from the char after the flag, to the end of
+                   rawCommandDetails */
+                extractedDetail = extractDetail(rawCommandDetail, flagIndexPositions[i] + flags[i].length());
+            } else {
+                // The extracted detail for non-last starts from the char after the flag, to index before the next flag
+                extractedDetail = extractDetail(
+                        rawCommandDetail,
+                        flagIndexPositions[i] + flags[i].length(),
+                        flagIndexPositions[i + 1]);
+            }
+            extractedCommandDetails.add(extractedDetail.trim());
+        }
+        return extractedCommandDetails;
+    }
+
+    private void checkForClientListIndexOutOfBounds(int clientIndex) throws InvalidIndexException {
+        if (clientIndex < 0 || clientIndex > clientList.getCurrentListSize() - 1) {
+            throw new InvalidIndexException(EXCEPTION);
+        }
+    }
+
+    private boolean isFlagPresent(int flagIndexPosition) {
+        return (flagIndexPosition != -1);
```

Comment:
Perhaps the '-1' can be represented with a constant. That would be more meaningful for the people reading the code.